temporal_sqlalchemy/session.py

```diff
@@ -7,20 +7,15 @@
 
 from temporal_sqlalchemy.bases import TemporalOption, Clocked
 from temporal_sqlalchemy.metadata import (
-    get_session_metadata,
-    set_session_metadata
+    STRICT_MODE_KEY,
+    CHANGESET_STACK_KEY,
+    IS_COMMITTING_KEY,
+    IS_VCLOCK_UNCHANGED_KEY,
 )
 
 
-CHANGESET_KEY = 'changeset'
-IS_COMMITTING_KEY = 'is_committing'
-IS_VCLOCK_UNCHANGED = 'is_vclock_unchanged'
-
-
-def _reset_flags(metadata):
-    metadata[CHANGESET_KEY] = {}
-    metadata[IS_COMMITTING_KEY] = False
-    metadata[IS_VCLOCK_UNCHANGED] = True
+def get_current_changeset(session):
+    return session.info[CHANGESET_STACK_KEY][-1]
```

Comment:
Can this ever be called if the list is empty?

If so, you'll get an `IndexError` exception.

Reply:
I think it could be possible in one code path.  Thanks for pointing that out.

Reply:
Okay.  Basically it shouldn't happen, but I could imagine some edge cases where the metadata isn't initiated because no temporal changes happened.  However, the stack should never be empty, so it will still fail on that.  I added an assertion to be very clear what the expectation is.  I think this is the appropriate behavior because it would indicate something is deeply wrong.  I will monitor for the warning.